server/src/main/java/pl/edu/agh/server/security/SecurityConfig.java

```diff
@@ -0,0 +1,55 @@
+package pl.edu.agh.server.security;
+
+import javax.sql.DataSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.provisioning.JdbcUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+
+@Configuration
+@EnableWebSecurity(debug = true)
+public class SecurityConfig {
+
+  private final String secret;
+
+  public SecurityConfig(@Value("${jwt.secret}") String secret) {
+    this.secret = secret;
+  }
+
+  @Bean
+  public SecurityFilterChain filterChain(HttpSecurity http,
+      AuthenticationManager authenticationManager,
+      UserDetailsService userDetailsService) throws Exception {
+
+    http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+
+    return http.authorizeRequests(
+            authorizeRequests -> authorizeRequests.requestMatchers("/tutor/**")
```

Comment:
tu rozumiem ze jak ida requesty na /tutor/** to wszystkie musza byc zautoryzowane?
Jesli tak to czy nie wystarczy nam tylko jedna rola uzytkownika?

Reply:
spring security wymaga roli dlatego jest wgl, standardowo jesli ktos nieporzadany nawet założyłby sobie konto to sprawdzimy czy może widzieć i nadamy mu uprawnienia